### PR TITLE
Adding Post Delete View

### LINKED ISSUE #99: Adding Post Delete View
- Adding the PostDeleteView class
- Adding the post confirm delete html file
- Adding tests to the delete view

### PR DESCRIPTION
**Pull request by AlonShakaroff**
- Adding the PostDeleteView class
- Adding the post confirm delete Html file
- Adding tests to the delete view
- Removed redundant spacing in tests/tests.py
- Adding 3 dots dropdown menu for each post for post actions.

fix #99 

image

image

image

### CODE REVIEW

File: posts/templates/posts/post_confirm_delete.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{% block styles %}
+    <link rel="stylesheet" type="text/css" href="{% static 'posts/forms.css' %}">
+{% endblock styles %}
+{% block content %}
+    <div class="content-section">
+        <form method="POST" enctype="multipart/form-data">
+            {% csrf_token %}
+            <fieldset class="form-group">
+                <legend class="border-bottom text-center mx-auto">Delete Post</legend>
+                <h2>Are you sure you want to delete the post?</h2>
```
Comment (MatanP12) [inline posts/templates/posts/post_confirm_delete.html:12]:
```suggestion
                <h2>Are you sure you want to delete this post?</h2>
```

File: posts/templates/posts/post_confirm_delete.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            </fieldset>
+            <div class="form-group">
+                <button class="btn btn-outline-danger" type="submit">Yes, Delete</button>
```
Comment (MatanP12) [inline posts/templates/posts/post_confirm_delete.html:15]: I think it should be something more like 

```suggestion
                  <button class="btn btn-outline-danger" type="submit">Confirm</button>
```

**Review by MatanP12 — CHANGES_REQUESTED** [review]

File: posts/templates/posts/post_confirm_delete.html (hunk 1 above)
Comment (AlonShakaroff) [reply posts/templates/posts/post_confirm_delete.html]: Done 👍

File: posts/templates/posts/post_confirm_delete.html (hunk 2 above)
Comment (AlonShakaroff) [reply posts/templates/posts/post_confirm_delete.html:15]: I think that "Yes, Delete" is a more suitable answer to the question: "Are you sure you want to delete this post?" than "Confirm".

File: posts/views.py [hunk 3]
```diff
@@ -28,6 +29,20 @@ class PostDetailView(DetailView):
...
+    # Django's default template name - posts/post_confirm_delete.html
+    context_object_name = 'post'
+    success_url = '/'
+
+    def test_func(self):
+        post = self.get_object()
+        if self.request.user == post.author:
+            return True
+        else:
+            return False
```
Comment (MatanP12) [inline posts/views.py:43]:
```suggestion
        return self.request.user == post.author:
```

**Review by MatanP12 — CHANGES_REQUESTED** [review]

File: posts/views.py (hunk 3 above)
Comment (AlonShakaroff) [reply posts/views.py]: Got you, fixed it here and also in the update view. 👍

**Review by MatanP12 — DISMISSED** [review]
LGTM!

**Review by yulisuliman — DISMISSED** [review]
LGTM

**Review by natifridman — DISMISSED** [review]

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by natifridman — APPROVED** [review]

**Pull request merged**